src/vellum/workflows/state/context.py

```diff
@@ -55,10 +67,63 @@ def execution_context(self) -> ExecutionContext:
     def generated_files(self) -> Optional[dict[str, str]]:
         return self._generated_files
 
+    @property
+    def enable_monitoring(self) -> bool:
+        """Whether workflow monitoring is enabled."""
+        return self._enable_monitoring or False
+
     @cached_property
     def node_output_mocks_map(self) -> Dict[Type[BaseOutputs], List[MockNodeExecution]]:
         return self._node_output_mocks_map
 
+    def get_monitoring_url(self, span_id: str) -> Optional[str]:
+        """
+        Generate the monitoring URL for this workflow execution.
+
+        Args:
+            span_id: The span ID from the workflow execution result.
+
+        Returns:
+            The URL to view execution details in Vellum UI, or None if monitoring is disabled.
+        """
+        if not self.enable_monitoring:
+            return None
+
+        def is_localhost_3000_running():
```

Comment:
[blocking] instead of `is_localhost_3000_running`, you could get the url from `self._context.vellum_client`